src/wallet/wallet.h

```diff
@@ -31,7 +31,8 @@
 #include <boost/shared_ptr.hpp>
 #include <boost/thread.hpp>
 
-extern CWallet* pwalletMain;
+typedef CWallet* CWallet_ptr;
```

Comment:
Indeed, a shared_ptr or similar would be needed for when/if we add the ability to close wallets at runtime.